package.json

```diff
@@ -21,22 +21,23 @@
     "access": "public"
   },
   "devDependencies": {
-    "@tsd/typescript": "^5.2.2",
-    "@types/jest": "^26.0.24",
-    "@types/node": "^16.4.10",
-    "jest": "^26.6.3",
-    "jest-runner-tsd": "^6.0.0",
-    "prettier": "^3.0.3",
-    "ts-jest": "^26.5.6",
-    "typescript": "^4.3.5"
+    "@tsd/typescript": "^5.4.3",
+    "@types/jest": "^29.5.12",
+    "@types/node": "^16.18.94",
+    "jest": "^29.7.0",
+    "prettier": "^3.2.5",
+    "ts-jest": "^29.1.2",
+    "tstyche": "^1.1.0",
```

Comment:
We were previously using [tsd-lite](https://github.com/mrazauskas/tsd-lite) along with `jest-runner-tsd` to run the type checks.  I noticed some issues with the type checks along with the typescript upgrade, and it appears that tsd-lite is now deprecated.  They recommend using [tstyche](https://github.com/tstyche/tstyche) instead, so I switched to this and updated the checks.

The only issue with tstyche is that it does not have a jest runner yet, so we have to run these checks with a separate command.  I updated the `package.json` to add a `check` command to run these tests, and updated the github workflow to add this step.